# Simple test of Pandas
* download test.csv file and try the problem below!

In [2]:
# import pandas
import pandas as pd

In [3]:
Data = pd.read_csv('test.csv')

In [4]:
# check the Data 
Data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1470 entries, 0 to 1469
Data columns (total 35 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   Age                       1470 non-null   int64 
 1   Attrition                 1470 non-null   object
 2   BusinessTravel            1470 non-null   object
 3   DailyRate                 1470 non-null   int64 
 4   Department                1470 non-null   object
 5   DistanceFromHome          1470 non-null   int64 
 6   Education                 1470 non-null   int64 
 7   EducationField            1470 non-null   object
 8   EmployeeCount             1470 non-null   int64 
 9   EmployeeNumber            1470 non-null   int64 
 10  EnvironmentSatisfaction   1470 non-null   int64 
 11  Gender                    1470 non-null   object
 12  HourlyRate                1470 non-null   int64 
 13  JobInvolvement            1470 non-null   int64 
 14  JobLevel                

In [6]:
# Test Just one data
Data.head(1)

Age Attrition BusinessTravel  DailyRate Department  DistanceFromHome  \
0   41       Yes  Travel_Rarely       1102      Sales                 1   

   Education EducationField  EmployeeCount  EmployeeNumber  ...  \
0          2  Life Sciences              1               1  ...   

   RelationshipSatisfaction StandardHours  StockOptionLevel  \
0                         1            80                 0   

   TotalWorkingYears  TrainingTimesLastYear WorkLifeBalance  YearsAtCompany  \
0                  8                      0               1               6   

  YearsInCurrentRole  YearsSinceLastPromotion  YearsWithCurrManager  
0                  4                        0                     5  

[1 rows x 35 columns]

## Simple Analysis
* Different types of datas
* There are 35 columns
* There are 1470 rows

## Problem 1
* Count the people who are Gender is `Female` and MonthlyRate is over `10,000`

In [7]:
# check the value of Gender data 
Data.Gender.value_counts()

Male      882
Female    588
Name: Gender, dtype: int64

In [8]:
# check the value of MonthlyRate data
Data.MonthlyRate.value_counts()

4223     3
9150     3
9558     2
12858    2
22074    2
        ..
14561    1
2671     1
5718     1
11757    1
10228    1
Name: MonthlyRate, Length: 1427, dtype: int64

In [10]:
# Answer (403)
len(Data[(Data['Gender'] == 'Female') & (Data['MonthlyRate'] >= 10000)])

403

## Problem 2
* Count the people who are EducationField is `Medical` and JobSatisfaction is over `3`

In [14]:
# check the value of EducationField data 
Data.EducationField.value_counts()

Life Sciences       606
Medical             464
Marketing           159
Technical Degree    132
Other                82
Human Resources      27
Name: EducationField, dtype: int64

In [15]:
# check the value of JobSatisfaction data 
Data.JobSatisfaction.value_counts()

4    459
3    442
1    289
2    280
Name: JobSatisfaction, dtype: int64

In [16]:
# Answer (282)
len(Data[(Data['EducationField'] == 'Medical') & (Data['JobSatisfaction'] >= 3)])

282